interpreter.py

```diff
@@ -118,83 +119,83 @@ def eval_condition(self, predicate, val1, val2):
             the two arguments, according to an ICmp predicate. '''
 
         assert isinstance(val1, Integer) and isinstance(val2, Integer)
-        if predicate == LLVMIntSLT:
```

Comment:
I'm not suggesting you do this now in this PR, but have a think about this: would it make more sense for each operation to have an `execute` method which accepts the interpreter state? then you wouldn't need this big switch and the code locality would be improved.

Only a suggestion at this point.